### PR TITLE
feat: optimize model service apis

### PR DESCRIPTION
**Pull request by bjwswang**
#### Which issue(s) this PR fixes

Fixes #

### CODE REVIEW

File: .github/workflows/graphql_sdk_generate.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Generate Graphql JS SDK
 
 on:
   push:
-    branches: [main]
+    branches: [chore]
```
Comment (Abirdcfly) [inline .github/workflows/graphql_sdk_generate.yml:5]: revert this?

Comment (bjwswang) [reply .github/workflows/graphql_sdk_generate.yml:5]: doing test my dev branch now . will revert this later.

File: apiserver/pkg/modelservice/modelservice.go [hunk 2]
```diff
@@ -156,180 +140,97 @@ func UpdateModelService(ctx context.Context, c dynamic.Interface, input generate
...
-			klog.V(5).Infof("add embedder modelservice: append only embedder modelService to list: %s", debugModelService(ms))
-
-			if r, ok := workerResource[v.Name]; ok {
-				klog.V(5).Infof("set embedder modelservice: %s resource: %+v", v.Name, r)
-				*ms.Resource = r.Resources
-				ms.ID = r.ID
-			}
+	// sort by creation timestamp
+	llmList.Nodes = append(llmList.Nodes, embedderList.Nodes...)
+	sort.Slice(llmList.Nodes, func(i, j int) bool {
```
Comment (bjwswang) [inline apiserver/pkg/modelservice/modelservice.go:215]: @0xff-dev  We can use the `CreationTimestamp` which has been pased into `ModelService `  for this `sort`.

File: apiserver/pkg/modelservice/modelservice.go [hunk 3]
```diff
@@ -338,213 +239,55 @@ func ListModelServices(ctx context.Context, c dynamic.Interface, input *generate
...
-	} else {
-		for idx := range llmList.Nodes {
-			v := llmList.Nodes[idx].(*generated.Llm)
-			if input.APIType != nil && *v.Type != *input.APIType {
-				continue
-			}
-			if input.ProviderType != nil && *v.Provider != *input.ProviderType {
+	// index is the actual result length
+	var index int
+	for _, service := range serviceList {
```
Comment (0xff-dev) [inline apiserver/pkg/modelservice/modelservice.go:249]: `map` is unordered, so the result may not be correct here.

File: apiserver/pkg/modelservice/modelservice.go (hunk 2 above)
Comment (bjwswang) [reply apiserver/pkg/modelservice/modelservice.go]: @0xff-dev  add a `newNodeList` for sort

File: apiserver/pkg/embedder/embedder.go [hunk 4]
```diff
@@ -31,10 +31,12 @@ import (
 	"github.com/kubeagi/arcadia/apiserver/graph/generated"
 	"github.com/kubeagi/arcadia/apiserver/pkg/common"
 	graphqlutils "github.com/kubeagi/arcadia/apiserver/pkg/utils"
+	"github.com/kubeagi/arcadia/apiserver/pkg/worker"
 	"github.com/kubeagi/arcadia/pkg/embeddings"
 	"github.com/kubeagi/arcadia/pkg/utils"
 )
 
+// Embedder2model convert unstructured `CR Embedder` to graphql model `Embedder`
```
Comment (Abirdcfly) [inline apiserver/pkg/embedder/embedder.go:39]:
```suggestion
// Embedder2model convert unstructured CR `Embedder` to graphql model `Embedder`
```
nit

File: apiserver/pkg/embedder/embedder.go [hunk 5]
```diff
@@ -248,6 +262,12 @@ func ListEmbedders(ctx context.Context, c dynamic.Interface, input generated.Lis
 
 	totalCount := len(us.Items)
 
+	// if pageSize is -1 which means unlimited pagesize,return all
+	if pageSize == common.UnlimitedPageSize {
+		page = 1
+		pageSize = totalCount
```
Comment (Abirdcfly) [inline apiserver/pkg/embedder/embedder.go:268]:
```
return &generated.PaginatedResult{
		TotalCount:  totalCount,
		HasNextPage: end < totalCount,
		Nodes:       result,
	}, nil
```

maybe in line 298, I guess this total is inaccurate, If keyword is not empty, the results have been filtered, but the total number has not changed. HasNextPage will always be true...

It should be a problem elsewhere.

File: apiserver/pkg/llm/llm.go [hunk 6]
```diff
@@ -31,10 +31,12 @@ import (
 	"github.com/kubeagi/arcadia/apiserver/graph/generated"
 	"github.com/kubeagi/arcadia/apiserver/pkg/common"
 	graphqlutils "github.com/kubeagi/arcadia/apiserver/pkg/utils"
+	"github.com/kubeagi/arcadia/apiserver/pkg/worker"
 	"github.com/kubeagi/arcadia/pkg/llms"
 	"github.com/kubeagi/arcadia/pkg/utils"
 )
 
+// LLM2model convert unstructured `CR LLM` to graphql model `Llm`
```
Comment (Abirdcfly) [inline apiserver/pkg/llm/llm.go:39]:
```suggestion
// LLM2model convert unstructured CR `LLM` to graphql model `Llm`
```

nit

**Review by 0xff-dev — APPROVED** [review]

File: apiserver/pkg/llm/llm.go [hunk 7]
```diff
@@ -44,14 +46,26 @@ func LLM2model(ctx context.Context, c dynamic.Interface, obj *unstructured.Unstr
...
 	status := common.GetObjStatus(llm)
 	message := string(condition.Message)
-
-	llmType := string(llm.Spec.Type)
-	provider := string(llm.Spec.Provider.GetType())
+	// Use worker's status&message if LLM's provider is `Worker`
+	if llm.Spec.Provider.GetType() == v1alpha1.ProviderTypeWorker {
+		w, err := worker.ReadWorker(ctx, c, llm.Name, llm.Namespace)
+		if err == nil {
+			if w.Status != nil {
```
Comment (Abirdcfly) [inline apiserver/pkg/llm/llm.go:61]: apiserver/pkg/embedder/embedder.go may need also check `w.Status != nil`

**Review by Abirdcfly — APPROVED** [review]

**Pull request merged**